modules/grizzly/src/main/java/org/glassfish/grizzly/filterchain/FilterChainContext.java

```diff
@@ -926,7 +927,8 @@ public String toString() {
         sb.append("connection=").append(getConnection());
         sb.append(", closeable=").append(getCloseable());
         sb.append(", operation=").append(getOperation());
-        sb.append(", message=").append(String.valueOf(getMessage()));
+        // message can be null, then valueOf(char[]) would be executed -> NPE
+        sb.append(", message=").append(String.valueOf(Object.class.cast(getMessage())));
```

Comment:
This produces something like `[C@78e94dcf` for non null message :disappointed:

Reply:
Which is probably correct as we don't want to see gigabyte in logs. Also usually the message is a Buffer instance, not char[]. char[] is probably never used as the message. To me it seems that generics were used wrong here as it can be anything but not anything is usable.

Reply:
The char[] is used as the most concrete valueOf method variant compatible with StringBuilder.append.
Such test would be probably misleading, but yeah, I can do some.